### ISSUE TITLE
Vulnerabilities detected on jenkins image

### ISSUE THREAD

**Issue by gabrilabs75** [jenkinsci/docker#1062]
Hi all,
these vulnerabilities were detected on latest jenkins image present on docker hub: ( 2.60.3-alpine)

CVE-2017-8816 
CVE-2017-8818 
Critical
curl
7.56.1-r0                                 fixed version  7.57.0-r0

CVE-2018-0500 
Critical
curl
7.56.1-r0                                 fixed version 7.61.0-r0

CVE-2018-1000005 
CVE-2018-1000007 
Critical
curl
7.56.1-r0                                 fixed version 7.58.0-r0

CVE-2018-1000120 
CVE-2018-1000122 
Critical
curl
7.56.1-r0                                 fixed version 7.59.0-r0

CVE-2018-1000300 
CVE-2018-1000301 
Critical
curl
7.56.1-r0                                fixed version 7.60.0-r0

CVE-2018-14618 
CVE-2018-16839          
CVE-2018-16840 
CVE-2018-16842 
Critical
curl
7.56.1-r0                             fixed version 7.61.1-r1

CVE-2018-17456 
Critical
git
2.11.3-r0                             fixed version 2.11.3-r2

CVE-2018-14599 
CVE-2018-14600 
Critical
libx11
1.6.4-r0                              fixed version 1.6.6-r0

CVE-2017-5461 
Critical
nss
3.27.2-r1                          fixed version 3.28.4-r0

CVE-2018-2938 
Critical
openjdk8
8.121.13-r0                      fixed version 8.181.13-r0

CVE-2018-3183 
Critical
openjdk8
8.121.13-r0                     fixed version  8.191.12-r0

CVE-2017-10989 
Critical
sqlite
3.15.2-r0                         fixed version 3.15.2-r1

CVE-2016-9841 
Critical
zlib 1.2.8-r2                    fixed version 1.2.11-r0

CVE-2016-9843 
Critical
zlib
1.2.8-r2                          fixed version 1.2.11-r0

We are trying to install a custom repository with some interesting features for our consumers.
As a registry we installed Harbor with "Trivy" add-on as vulnerability scanner.
When I pushed the latest image of jenkins from docker hub to our registry "Trivy" detected the vulnerabilities above marked as "Critical".

Could you let us know when you fixed the image with the correct libs and packages version or if we could use another image?

Thanks in advance.
Regards,
Gabriele

**Comment by timja** [comment]
@gabrilabs75 that image has been deprecated for over 2 years, see warning: https://hub.docker.com/_/jenkins

Please see the tags for the jenkins/jenkins image:
https://hub.docker.com/r/jenkins/jenkins

We're mostly dependent on upstream maintainers for minimal CVEs,

alpine, slim and centos are your best bet I believe.

**Comment by gabrilabs75** [comment]
@timja thanks Tim! I didn't notice that...
Anyway I tried to push this version **2.263.1-lts-alpine** and Trivy found these vulnerabilities marked as "High"

CVE-2020-8231 
High
curl
7.69.1-r1               fixed on 7.69.1-r2

CVE-2020-8285 
CVE-2020-8286 
High
curl
7.69.1-r1               fixed on 7.69.1-r3


Could it depend on the "starting" operating system ?

**Comment by daniel-beck** [comment]
Note that for system components, we just deliver what upstream provides at the time the image is created. While there may be some choices we can make about the default and available variants (e.g. https://github.com/jenkinsci/docker/issues/1033), upstream is generally far better equipped to address system-level issues than the Jenkins project.

In this case, Alpine addressed those CVEs in 3.12.3: https://www.alpinelinux.org/posts/Alpine-3.12.3-released.html

Which, according to https://alpinelinux.org/downloads/, was

> Released Dec 16, 2020

while 2.263.1 was released Dec 3. LTS releases are on a roughly monthly release cadence, if you need more than that, you will need to do it yourself.

**Comment by MarkEWaite** [comment]
Alpine 3.12.3 is included in the Jenkins 2.263.3 Docker image as reported by:

```
mwaite@mark-pc2:~$ docker run --rm -i -t jenkins/jenkins:2.263.3-alpine cat /etc/os-release
Digest: sha256:fa0b0f06383944877c8105ad8e7edcac29f05583cef8b0d26cd8c9586e0cb667
Status: Downloaded newer image for jenkins/jenkins:2.263.3-alpine
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.12.3
PRETTY_NAME="Alpine Linux v3.12"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"
```
